api/v1/handlers.go

```
@@ -1743,37 +1730,58 @@ func (s *apiV1) handleGetSystemConfig(c echo.Context, u *util.User) error {
 }
 
 type minerResp struct {
-	Addr            address.Address `json:"addr"`
-	Name            string          `json:"name"`
-	Suspended       bool            `json:"suspended"`
-	SuspendedReason string          `json:"suspendedReason,omitempty"`
-	Version         string          `json:"version"`
+	Addr            address.Address       `json:"addr"`
+	Name            string                `json:"name"`
+	Suspended       bool                  `json:"suspended"`
+	SuspendedReason string                `json:"suspendedReason,omitempty"`
+	Version         string                `json:"version"`
+	ChainInfo       *miner.MinerChainInfo `json:"chain_info"`
 }
 
 // handleAdminGetMiners godoc
 // @Summary      Get all miners
-// @Description  This endpoint returns all miners
-// @Tags         public,net
+// @Description  This endpoint returns all miners. Note: value may be cached
+// @Tags         admin,net
 // @Produce      json
 // @Success      200  {object}  string
 // @Failure      400           {object}  util.HttpError
 // @Failure      500           {object}  util.HttpError
-// @Router       /public/miners [get]
+// @Router       /admin/miners/ [get]
 func (s *apiV1) handleAdminGetMiners(c echo.Context) error {
+	ctx := context.TODO()
+
+	// Cache the Chain Lookup for this miner, looking up if it doesnt exist / is expired
+	key := util.CacheKey(c, nil)
+	cached, ok := s.extendedCacher.Get(key)
+	if ok {
+		out, ok := cached.([]minerResp)
+		if ok {
```

Same as below, we might want to return an error if `!ok` here